addons/purchase_stock/models/stock.py

```diff
@@ -37,18 +37,8 @@ def _prepare_merge_move_sort_method(self, move):
     def _get_price_unit(self):
         """ Returns the unit price for the move"""
         self.ensure_one()
-        if self.purchase_line_id and self.product_id.id == self.purchase_line_id.product_id.id:
-            line = self.purchase_line_id
-            order = line.order_id
-            price_unit = line.price_unit
-            if line.taxes_id:
-                price_unit = line.taxes_id.with_context(round=False).compute_all(price_unit, currency=line.order_id.currency_id, quantity=1.0)['total_excluded']
-            if line.product_uom.id != line.product_id.uom_id.id:
-                price_unit *= line.product_uom.factor / line.product_id.uom_id.factor
-            if order.currency_id != order.company_id.currency_id:
-                price_unit = order.currency_id._convert(
-                    price_unit, order.company_id.currency_id, order.company_id, self.date, round=False)
-            return price_unit
+        if self.purchase_line_id and self.product_id == self.purchase_line_id.product_id:
```

Comment:
TBH, I'm not sure why we need this. I have tried to follow git blame, but there have been 2 major refactorings (removal of procurement.order + split in `purchase_stock`), and I don't find the exact reason for adding it.

I think this is not needed at all, as the price is already put in `price_unit` field of `stock.move` in the moment of picking creation, and you can't manually change this, so there's no option of having an altered value in the middle.

Reply:
@pedrobaeza Sometimes you need a different value for stock move price and purchase price unit. IAS2 is pretty explicit on this. So you need 2 separate methods. If you want to refactor out to one method for the default behaviour then go ahead, but don't point it to the same method please.

A very simple example is a volume rebate paid monthly which is generally always hit, that must be reflected in the price of inventory but would not be reflected in purchase price.

Reply:
@gdgellatly you already have a different value, as you have a field `price_unit` in `stock.move` that is taken for the stock valuation. But this method fills again that value with the same PO price that was dumped here before.

Reply:
@pedrobaeza We will soon find out, maybe it isn't a problem. There used to be another method here, get_price_unit_invoice or some such that seems to have gone which was the one I was thinking of.